src/poetry/inspection/lazy_wheel.py

```diff
@@ -0,0 +1,755 @@
+"""Lazy ZIP over HTTP"""
+
+from __future__ import annotations
+
+import abc
+import io
+import logging
+import re
+
+from bisect import bisect_left
+from bisect import bisect_right
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import BinaryIO
+from typing import ClassVar
+from typing import cast
+from urllib.parse import urlparse
+from zipfile import BadZipFile
+from zipfile import ZipFile
+
+from packaging.metadata import parse_email
+from requests.models import CONTENT_CHUNK_SIZE
+from requests.models import HTTPError
+from requests.models import Response
+from requests.status_codes import codes
+
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+    from collections.abc import Iterator
+
+    from packaging.metadata import RawMetadata
+    from requests import Session
+
+    from poetry.utils.authenticator import Authenticator
+
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPRangeRequestUnsupported(Exception):
+    """Raised when the remote server appears unable to support byte ranges."""
+
+
+class UnsupportedWheel(Exception):
+    """Unsupported wheel."""
+
+
+class InvalidWheel(Exception):
+    """Invalid (e.g. corrupt) wheel."""
+
+    def __init__(self, location: str, name: str):
```

Comment:
```suggestion
    def __init__(self, location: str, name: str) -> None:
```

Reply:
Good catch. I have always thought that mypy catches this but it looks like there is an exception for `__init__`. 😲